source/api_cc/include/DeepPot.h

```diff
@@ -629,6 +629,36 @@ class DeepPotModelDevi {
             const std::vector<std::string>& file_contents =
                 std::vector<std::string>());
 
+/**
+   * @brief Evaluate the energy, force and virial by using these DP models.
+   * @param[out] all_ener The system energies of all models.
+   * @param[out] all_force The forces on each atom of all models.
+   * @param[out] all_virial The virials of all models.
+   * @param[in] coord The coordinates of atoms. The array should be of size
+   *nframes x natoms x 3.
+   * @param[in] atype The atom types. The list should contain natoms ints.
+   * @param[in] box The cell of the region. The array should be of size nframes
+   *x 9.
+   * @param[in] fparam The frame parameter. The array can be of size :
+   * nframes x dim_fparam.
+   * dim_fparam. Then all frames are assumed to be provided with the same
+   *fparam.
+   * @param[in] aparam The atomic parameter The array can be of size :
+   * nframes x natoms x dim_aparam.
+   * natoms x dim_aparam. Then all frames are assumed to be provided with the
+   *same aparam. dim_aparam. Then all frames and atoms are provided with the
+   *same aparam.
+   **/
+  template <typename VALUETYPE>
+  void compute(std::vector<ENERGYTYPE>& all_ener,
+               std::vector<std::vector<VALUETYPE> >& all_force,
+               std::vector<std::vector<VALUETYPE> >& all_virial,
+               const std::vector<VALUETYPE>& coord,
```

Comment:
I am consued that why the variables `coord` and `atype` are not the same as they are in the implementation `dcoord_`, `datype` @njzjz . I just follow the convention. Is it a typo or something else?

Reply:
The convention exists in the initial version: https://github.com/deepmodeling/deepmd-kit/blame/e86cf20b4a9ad6ae629ec16c566ed3b086cb4b75/source/lib/src/NNPInter.cc
where dcoord is a vector and coord is a tensor. @wanghan-iapcm may answer.